### DISCUSSION TITLE
A bug when choosing keys and duplicating them with the CTRL button

### DISCUSSION THREAD

**Discussion by beeheemooth** [opentoonz/opentoonz#6802] [General]
### Issue Summary

Let's look at the following:
When selecting keys, the Ctrl key adds a new key to the selected ones (or deselects it, if it is already selected) 

Copying of keys is also carried out when the Ctrl button is pressed and the cursor is dragged, but at the same time, if all the keys are selected, then the key for which you pull stops being selected, and only the rest of the highlighted keys are copied. 

Apparently, the order of operations is playing a role here. 

In this case, selecting / deselecting a key is a priority, because it happens as soon as you click on it, and in theory, it should happen when you release the mouse button. 

The operation is canceled if you move the mouse, but did not release the button, because in this case the keys are duplicated and moved. What do you think about this?

### Screenshot or Video Reference

![ezgif-1-d8ef67f3e31d](https://user-images.githubusercontent.com/25724133/69459821-54056680-0d7b-11ea-846a-eeabffc41779.gif)

### System Information
* **OpenToonz Version**: latest
* **Operating System**:  Win 8.1

**Comment by RodneyBaker** [comment]
From what I can tell here using the Control key copies (and moves) everything EXCEPT the key that has focus.  Using the Shift key copies (and moves) all the keys.

Regarding the priority aspect of the processing of key copying... this may be an issue similar to the one where files were deleted due to 'moving' them before they could properly copied/processed, resulting in inadvertent removal.  @manongjohn can possibly speak to that aspect as he implemented the fix for the issue with file loss.

As for this issue in general I'm not sure whether it is working as intended or a bug.
More investigation/information needed.
